dvc/lock.py

```diff
@@ -3,58 +3,118 @@
 from __future__ import unicode_literals
 
 import os
+import hashlib
 import time
-import zc.lockfile
-
-from dvc.exceptions import DvcException
+from datetime import timedelta
 
+from funcy.py3 import lkeep
 
-class LockError(DvcException):
-    """Thrown when unable to acquire the lock for dvc repo."""
+from dvc.exceptions import DvcException
+from dvc.utils import makedirs
+from dvc.utils.compat import is_py3
 
 
-class Lock(object):
-    """Class for dvc repo lock.
+DEFAULT_TIMEOUT = 5
 
-    Args:
-        dvc_dir (str): path to the directory that the lock should be created
-            in.
-        name (str): name of the lock file.
-    """
 
-    LOCK_FILE = "lock"
-    TIMEOUT = 5
+class LockError(DvcException):
+    """Thrown when unable to acquire the lock for dvc repo."""
 
-    def __init__(self, dvc_dir, name=LOCK_FILE):
-        self.lock_file = os.path.join(dvc_dir, name)
-        self._lock = None
 
-    def _do_lock(self):
-        try:
-            self._lock = zc.lockfile.LockFile(self.lock_file)
-        except zc.lockfile.LockError:
-            raise LockError(
-                "cannot perform the cmd since DVC is busy and "
-                "locked. Please retry the cmd later."
-            )
+if is_py3:
+    import flufl.lock
+
+    class Lock(flufl.lock.Lock):
+        """Class for dvc repo lock.
+
+        Args:
+            lockfile (str): the lock filename
+                in.
+            lifetime (int | timedelta): hold the lock for so long.
+            tmp_dir (str): a directory to store claim files.
+        """
+
+        def __init__(self, lockfile, lifetime=None, tmp_dir=None):
```

Comment:
Thanks for setting a long default lifetime. Don't see this one used anywhere now
```suggestion
        def __init__(self, lockfile, tmp_dir=None):
```

Reply:
For the record: will remove this myself right after merge to move along faster.